ibis/pyspark/tests/test_timecontext.py

```diff
@@ -9,7 +14,408 @@
+            'count_1h', F.count(spark_table['value']).over(spark_window_1h)
+        )
+        .withColumn(
+            'count_2h', F.count(spark_table['value']).over(spark_window_2h)
+        )
+        .toPandas()
+    )
+    expected = expected[
+        expected.time.between(*(t.tz_convert(None) for t in context))
+    ].reset_index(drop=True)
+    tm.assert_frame_equal(result_pd, expected)
+
+
+@pytest.mark.xfail(
+    reason='Issue #2457 Adjust context properly for mixed rolling window,'
+    ' cumulative window and non window ops',
+    strict=True,
+)
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [
+        (
+            [('trailing', 1), ('cumulative', 1)],
+            [(-3600, 0), (Window.unboundedPreceding, 0)],
+        )
+    ],
+    indirect=['ibis_window'],
+)
+def test_rolling_with_cumulative_window(client, ibis_window, spark_range):
+    """ Test context adjustment for rolling window and cumulative window
+
+    cumulative window should calculate only with in user's context,
+    while rolling window should calculate on expanded context.
+    For a rolling window of 1 day,
+    time       value
+    2020-01-01   a
+    2020-01-02   b
+    2020-01-03   c
+    2020-01-04   d
+    with context = (2020-01-02, 2002-01-03), count will be:
+    time       value  roll_count cum_count
+    2020-01-02   b      2            1
+    2020-01-03   c      2            2
+    """
+    table = client.table('time_indexed_table')
+    context = (
+        pd.Timestamp('20170102 07:00:00', tz='UTC'),
+        pd.Timestamp('20170105', tz='UTC'),
+    )
+    result_pd = table.mutate(
+        count_1h=table['value'].count().over(ibis_window[0]),
+        count_cum=table['value'].count().over(ibis_window[1]),
+    ).execute(timecontext=context)
+
+    spark_table = table.compile()
+    spark_window_1h = (
+        Window.partitionBy('key')
+        .orderBy(F.col('time').cast('long'))
+        .rangeBetween(*spark_range[0])
+    )
+    spark_window_cum = (
+        Window.partitionBy('key')
+        .orderBy(F.col('time').cast('long'))
+        .rangeBetween(*spark_range[1])
+    )
+    expected = (
+        spark_table.withColumn(
+            'count_1h', F.count(spark_table['value']).over(spark_window_1h)
+        )
+        .withColumn(
+            'count_cum', F.count(spark_table['value']).over(spark_window_cum)
+        )
+        .toPandas()
+    )
+    expected = expected[
+        expected.time.between(*(t.tz_convert(None) for t in context))
+    ].reset_index(drop=True)
+    tm.assert_frame_equal(result_pd, expected)
+
+
+@pytest.mark.xfail(
+    reason='Issue #2457 Adjust context properly for mixed rolling window,'
+    ' cumulative window and non window ops',
+    strict=True,
+)
+@pytest.mark.parametrize(
+    ('ibis_window', 'spark_range'),
+    [([('trailing', 1)], [(-3600, 0)])],
+    indirect=['ibis_window'],
+)
+def test_rolling_with_non_window_op(client, ibis_window, spark_range):
```

Comment:
Why are these not n tess_window.py?

Reply:
We need to make sure time context is well adjusted

Reply:
Actually I think we should move window tests to test_window_context_adjustment.py

Reply:
Time context is the main thing that we focus in such combined windows.  There is no time context involved in `test_window.py`

Reply:
I see. I think `test_timecontext.py` is bit of misnomer because this sounds like testing the class timecontext or related functions, but you are actually testing lots of the functionality in windowing. Therefore I'd suggest move these tests to another file so that it's clearer.

Reply:
Fine with me! I leave `test_timecontext` only to test time context related functions, all window tests without time context in `test_window.py`, and all window tests with time context in `test_window_context_adjustment.py` as you suggested.